easybuild/easyblocks/g/go.py

```diff
@@ -61,7 +61,7 @@ def install_step(self):
 
         # $GOROOT_FINAL only specifies the location of the final installation, which gets baked into the binaries
         # the installation itself is *not* done by the all.bash script, that needs to be done manually
-        cmd = "GOROOT_FINAL=%s ./all.bash" % self.installdir
+        cmd = "GOROOT_BOOTSTRAP=%s GOROOT_FINAL=%s ./all.bash" % (os.environ.get('EBROOTGO'),self.installdir)
```

Comment:
The 1.5 version of Go needs go to be compiled. Hence it needs the 1.4.2
module loaded to be built. (I've just realised that I would need to change
the Easyconfigs to avoid having both versions loaded due to the dependency)
The 1.4.2 only needs gcc to be compiled, so the EBROOTGO should be empty

On Mon, Sep 7, 2015 at 6:37 PM, Kenneth Hoste <notifications@github.com>
wrote:

> In easybuild/easyblocks/g/go.py
> <https://github.com/hpcugent/easybuild-easyblocks/pull/683#discussion_r38873196>
> :
>
> > @@ -61,7 +61,7 @@ def install_step(self):
> >
> >          # $GOROOT_FINAL only specifies the location of the final installation, which gets baked into the binaries
> >          # the installation itself is *not* done by the all.bash script, that needs to be done manually
> > -        cmd = "GOROOT_FINAL=%s ./all.bash" % self.installdir
> > +        cmd = "GOROOT_BOOTSTRAP=%s GOROOT_FINAL=%s ./all.bash" % (os.environ.get('EBROOTGO'),self.installdir)
>
> looks good, but you should use the get_software_root function rather than
> hardcoding $EBROOTGO
>
> or, actually, how can EBROOTGO be defined in the environment when you're
> installing Go?!
>
> maybe you mean this?
>
> cmd = "GOROOT_BOOTSTRAP=%(installdir)s GOROOT_FINAL=%(installdir)s ./all.bash" % {'installdir': self.installdir}
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hpcugent/easybuild-easyblocks/pull/683/files#r38873196>
> .
>



-- 
--
Carles Fenoy